src/lib.rs

```diff
@@ -1877,6 +1877,13 @@ impl Connection {
                     // TODO: in the future we might want to buffer other types
                     // of undecryptable packets as well.
                     let pkt_len = b.off() + payload_len;
+
+                    // Don't add to the queue if this packet is not decryptable
+                    // because of the mismatched size.
+                    if b.len() < pkt_len {
```

Comment:
This check should happen sooner I think, imediately after we decode the payload length (https://github.com/cloudflare/quiche/blob/master/src/lib.rs#L1824-L1835), so we can exit early (though other packets with invalid payload length would get dropped anyway).

Since we don't have `pkt_len` there, could be something like:
```rust
        if payload_len > b.cap() {
            return Err(drop_pkt_on_err(
                Error::InvalidPacket,
                self.recv_count,
                self.is_server,
                &self.trace_id,
            ));
        }
```

Reply:
Updated. This also changes `invalid_initial_payload()` test because the size is checked earlier.